app/main.py

```diff
@@ -1,3 +1,102 @@
-def shop_trip():
-    # write your code here
-    pass
+from app.customer import Customer
+from app.shop import Shop
+from app.car import Car
+
+
+from math import sqrt
+import json
+import datetime
+
+
+def shop_trip() -> None:
+    with open("app/config.json", "r") as file:
+        data = json.load(file)
+
+    fuel_price = data["FUEL_PRICE"]
+    check_data = datetime.datetime.now().strftime("%d/%m/%Y %H:%M:%S")
+
+    customers_list = [
+        Customer(customer["name"],
+                 customer["product_cart"],
+                 customer["location"],
+                 customer["money"])
+        for customer in data["customers"]
+    ]
+
+    shops_list = [
+        Shop(shop["name"],
+             shop["location"],
+             shop["products"])
+        for shop in data["shops"]
+    ]
+
+    cars_list = [
+        Car(car["car"]["brand"], car["car"]["fuel_consumption"])
+        for car in data["customers"]
+    ]
+
+    def product_cost(customer: Customer,
+                     shop: Shop,
+                     ) -> int | float:
+        milk_cost = customer.product_cart["milk"] * shop.products["milk"]
+        bread_cost = customer.product_cart["bread"] * shop.products["bread"]
+        butter_cost = customer.product_cart["butter"] * shop.products["butter"]
+        return milk_cost + bread_cost + butter_cost
+
+    def cost_trip(customer: Customer,
+                  car: Car,
+                  shop: Shop
+                  ) -> float | int:
+        location = sqrt((shop.location[0] - customer.location[0]) ** 2
+                        + (shop.location[1] - customer.location[1]) ** 2)
+        cost_fuel = location * fuel_price * car.fuel_consumption / 100
+        return round(cost_fuel * 2, 2)
+
+    def go_to_home(remainder: int or float) -> str:
```

Comment:
```suggestion
    def go_to_home(remainder: int | float) -> str:
```
Or Union[int, float]